### PR TITLE
feat(edit-ema): Trigger Workflow Actions #27159

### LINKED ISSUE #27159: Edit EMA: Add and Trigger Workflow Actions
### User Story
As a dotCMS user, I want to be able to trigger a workflow action from the editing page, so I can quickly take action on my content.

### Acceptance Criteria
- Add a split button to the toolbar
- Read the options from the workflow actions endpoints 
- Trigger the action with the fire workflow action endpoint 

### Proposed Objective
Core Features

### Proposed Priority 
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
N/A

### Assumptions & Initiation Needs
N/A

### Quality Assurance Notes & Workarounds 
N/A

### Sub-Tasks & Estimates
- Design split button 
- Add split button to toolbar 
- Read options from workflow actions endpoints 
- Trigger action with fire workflow action endpoint

### PR DESCRIPTION
**Pull request by rjvelazco**
### Feature
- Add and trigger workflow actions in edit EMA

#### Video

https://github.com/dotCMS/core/assets/72418962/655bf1f6-c602-4887-9e34-94a9065b46b5

#### Push Publish
https://github.com/dotCMS/core/assets/72418962/3f89e1d7-3d71-4dc8-b6a2-fe7df871421c

### CODE REVIEW

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts [hunk 1]
```diff
@@ -15,10 +15,14 @@ import { map, skip, take, takeUntil } from 'rxjs/operators';
...
     DotFavoritePageService,
+    DotHttpErrorManagerService,
     DotLanguagesService,
     DotPageLayoutService,
     DotPageRenderService,
-    DotPersonalizeService
+    DotPersonalizeService,
+    DotWorkflowActionsFireService,
+    DotWorkflowsActionsService,
+    DotWorkflowEventHandlerService
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts:25]: This is why modules are for no? Maybe... just a suggestion.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+@if (actions()) {
+<dot-workflow-actions
+    [loading]="loading()"
+    [actions]="actions()"
+    (actionFired)="handleActionTrigger($event)" />
+} @else {
+<p-button [loading]="true" [label]="'Loading' | dm" data-testId="loading-button" />
+}
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.html:8]:
```suggestion
@if (actions()) {
  <dot-workflow-actions
      [loading]="loading()"
      [actions]="actions()"
      (actionFired)="handleActionTrigger($event)" />
  } @else {
    <p-button [loading]="true" [label]="'Loading' | dm" data-testId="loading-button" />
  }
```

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+@Component({
+    selector: 'dot-edit-ema-workflow-actions',
+    standalone: true,
+    imports: [AsyncPipe, DotWorkflowActionsComponent, ButtonModule, DotMessagePipe],
+    providers: [
+        DotWorkflowActionsFireService,
+        DotWorkflowEventHandlerService,
+        DotWorkflowsActionsService,
+        DotHttpErrorManagerService
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.ts:37]: If you are injecting this here, why are you also doing it in the shell component?

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.spec.ts [hunk 4]
```diff
@@ -9,34 +9,45 @@ import { of } from 'rxjs';
...
     DotPersonalizeServiceMock,
     DotDevicesServiceMock,
     mockDotDevices,
     LoginServiceMock,
-    DotCurrentUserServiceMock
+    DotCurrentUserServiceMock,
+    dotcmsContentletMock
 } from '@dotcms/utils-testing';
 
+import { DotEditEmaWorkflowActionsComponent } from './components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component';
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.spec.ts:50]: Mock this... so you dont have to include alllllll it deps.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.spec.ts [hunk 5]
```diff
@@ -211,7 +227,17 @@ const createRouting = (permissions: { canEdit: boolean; canRead: boolean }) =>
...
+            mockProvider(DotContentTypeService),
+            mockProvider(DotWorkflowActionsFireService),
+            mockProvider(DotMessageDisplayService),
+            mockProvider(DotHttpErrorManagerService),
+            mockProvider(DotRouterService),
+            mockProvider(PushPublishService),
+            mockProvider(ApiRoot),
+            mockProvider(DotFormatDateService),
+            mockProvider(DotIframeService),
+            mockProvider(DotGlobalMessageService)
```
Comment (fmontes) [inline core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.spec.ts:240]: Like all this.

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.html (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.html:8]: @fmontes I can add the space in the `@if/@else`, the pre-commit removes them.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.ts (hunk 3 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.ts:37]: Forgot to remove them. Nice catch @fmontes!

File: core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts (hunk 1 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts:25]: I just let the services on the component that needs them.

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.spec.ts (hunk 5 above)
Comment (rjvelazco) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.spec.ts:240]: Done!

**Review by zJaaal — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Review by manuelrojas — APPROVED** [review]

File: core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.html (hunk 2 above)
Comment (fmontes) [reply core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-edit-ema-workflow-actions/dot-edit-ema-workflow-actions.component.html:8]: WHAT!!!! @oidacra we need to do something here 🤣

**Review by fmontes — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Pull request merged**